// VERILOG_OUTPUT: a.v

// regression test in bb conditions merge.

func generateNumbers() {
  var i int
  for i = 0; i < 10; ++i {
    var next_idx int = i + 1
    if (next_idx > 5) {
      next_idx = 0
    }
  }
  assert(i == 10)
}

func main() {
  // wrapping this function caused the fault.
  generateNumbers()
}

main()

compile()
writeHdl("a.v")
